content/en/docs/Advanced/revoking-secureboot-access.md

```diff
@@ -0,0 +1,337 @@
+systemd-cryptenroll --unlock-key-file=/tmp/random_keyfile --tpm2-public-key=/run/systemd/tpm2-pcr-public-key.pem --tpm2-public-key-pcrs=11 --tpm2-pcrs=7 --tpm2-signature=/run/systemd/tpm2-pcr-signature.json --tpm2-device=auto /dev/vda3
+
+echo "Removing the password slot"
+systemd-cryptenroll --wipe-slot=password /dev/vda2
+systemd-cryptenroll --wipe-slot=password /dev/vda3
+```
+
+The following command:
+
+```bash
+cryptsetup luksDump /dev/vda2 | grep "pcrs: "
+```
+
+should show output similar to this:
+
+```
+	tpm2-hash-pcrs:   7
+	tpm2-pubkey-pcrs: 11
+```
+
+which means the decryption is now bound again to PCR 7 (directly) and PCR 11 (indirectly).
+
+This command:
+
+```
+systemd-cryptenroll /dev/vda2
+```
+
+should only show `tmp2` (and not `password`)
+
+(same thing for `/dev/vda3`)
+
+You can now reboot to check if everything works correctly
+
+```bash
+reboot
+```
+
+#### Verifying that it worked
+
+- Check what keys are enrolled in UEFI db:
+
+```
+mokutil --list-enrolled --db | grep -E 'Issuer:|Subject:'
+```
+
+You should find your new certificate in the list
+
+- Check if the old certificate is in dbx:
+
+```
+mokutil --list-enrolled --dbx | grep -E 'Issuer:|Subject:'
+```
+
+You should find your old certificate in the list.
+
+- Check that the old images are not bootable
+
+Just reboot and when presented with the boot menu, select "fallback" or "recovery".
+Since those images haven't been upgraded yet, they are still signed with the old
+certificate. They should not boot and UEFI should show an error.
+
+
+#### Upgrading recovery
+
+After successfully booting to "active", you can now upgrade `recovery` to make
+can still boot if the next update goes wrong.
+
+Obviously the recovery image will also need to be signed with the new certificate.
+
+The upgrade command is:
+
+```
+kairos-agent upgrade --recovery --source oci:<YOUR_UPGRADE_IMAGE_HERE>
+```
+
+The "fallback" image will be upgraded on the next upgrade. Read the ["Container based"]({{< relref "../architecture/container" >}})
+document to understand more.
+
+## Scenario 2 - One specific image is no longer trusted
+
+In the previous scenario, we assumed that the signing certificate was compromised, which meant no image signed by that certificate should be bootable anymore.
+
+In this scenario, only one specific image is considered "bad". For example, an image shipped with a well known security vulnerability.
+In order to prevent this image from booting, its "hash" can be blacklisted.
+
+The process is exactly the same as above, but instead of enrolling a certificate in the `dbx` database, we will enroll an image hash.
+
+Follow the same instructions as in scenario 1, but skipping the steps where a new `db` key is created.
+
+The easiest way to find the offending `.efi` file and calculate its signature is to boot the image which we
+want to blacklist in a sandboxed environment.
```

Comment:
I'm not sure we should suggest this. Maybe I should show instructions for the extraction from an ISO. But it gets messy.

Reply:
yes indeed. Maybe on EFI build we should spit a file which contains info about the EFI image AND its ginature+hash and so on? So that can get archived somewhere in order to refer to it? So then you can just use that hash to blacklist it or know which signature it was used?

Reply:
also maybe suggest that you can boot with the newer install media, and calculate it via that, so you are on a safe environment? boot from livecd, mount efi partition, calculate signatures or whatever

Reply:
Producing the information at build time (your first suggestion) doesn't sound enough. Maybe they no longer have access to those files anymore (the build could have happened months ago).

I prefer your second suggestion because it doesn't require access to any old files. But I need to try that out and come up with instructions. Let me try.

Reply:
I added some instructions around this (not copy-paste things)